service/src/main/kotlin/org/opensearch/simpleschema/model/SchemaCompilationType.kt

```diff
@@ -0,0 +1,123 @@
+package org.opensearch.simpleschema.model
+
+import org.opensearch.common.io.stream.StreamInput
+import org.opensearch.common.io.stream.StreamOutput
+import org.opensearch.common.io.stream.Writeable
+import org.opensearch.common.xcontent.XContentParser
+import org.opensearch.common.xcontent.XContentFactory
+import org.opensearch.common.xcontent.ToXContent
+import org.opensearch.common.xcontent.XContentBuilder
+import org.opensearch.common.xcontent.XContentParserUtils
+import org.opensearch.simpleschema.SimpleSchemaPlugin.Companion.LOG_PREFIX
+import org.opensearch.simpleschema.util.fieldIfNotNull
+import org.opensearch.simpleschema.util.logger
+import org.opensearch.simpleschema.util.stringList
+
+/**
+ * This element represents a schema entity which is stored as a document in the DB
+ * The entity has a type, name, catalog, description and the actual GraphQL SDL as content
```

Comment:
has a group of GraphQL components such as : types, mutations, queries, subscriptions

Reply:
Added clarification in doc that IDs referencing these go in the entities list, should they be their own objects?